source/ast/ASTSerializer.cpp

```diff
@@ -198,6 +215,20 @@ void ASTSerializer::visit(const T& elem, bool inMembersArray) {
     else if constexpr (std::is_base_of_v<Statement, T>) {
         writer.startObject();
         write("kind", toString(elem.kind));
+        if (elem.syntax != nullptr) {
+            write("source_file_start",
```

Comment:
There is a lot of duplication here. Please extract this logic out into a helper method that can operate on a SyntaxNode so it's not repeated several times.